src/components/fieldset/fieldset.style.ts

```diff
@@ -49,6 +53,31 @@ const LegendContainerStyle = styled.div<StyledFieldsetProps>`
     font-weight: 600;
     line-height: 24px;
     margin-right: 4px;
+
+    ${({ isRequired }) =>
+      isRequired &&
+      css`
+        ::after {
+          content: "*";
+          line-height: 24px;
+          color: var(--colorsSemanticNegative500);
+          font-weight: 700;
+          margin-left: var(--spacing100);
+          position: relative;
+          top: 1px;
+          left: -4px;
+        }
+      `}
+
+    ${({ isOptional }) =>
+      isOptional &&
+      css`
+        ::after {
+          content: "(optional)";
+          font-weight: 350; //TODO: (tokens) use token var(--fontWeights400) - FE-6022
+          margin-left: 4px;
```

Comment:
**suggestion(non-blocking):** The DS token for 4px is `spacing050` if you wanted to use that here as you've used the DS token for margin-left on the `::after` styling. Just thinking for consistency.